Compatibility to JDK 8 and JDK 9 at the same time?

Hi, is the next version supposed to support JDK9 and JDK8? I'm currently experiencing this exception when running the master on JDK8. Or is there a 'special' way i need to build the jar? 

```java
Caused by: java.lang.NoClassDefFoundError: java/lang/StackWalker$Option
	at javassist.util.proxy.DefineClassHelper$SecuredPrivileged$1.<init>(DefineClassHelper.java:67)
	at javassist.util.proxy.DefineClassHelper$SecuredPrivileged.<clinit>(DefineClassHelper.java:38)
	at javassist.util.proxy.DefineClassHelper.<clinit>(DefineClassHelper.java:177)
	at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:128)
	at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:552)
	at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:537)
	at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:473)
	at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:444)
	at org.hibernate.proxy.pojo.javassist.JavassistProxyFactory.postInstantiate(JavassistProxyFactory.java:75)
	at org.hibernate.tuple.entity.PojoEntityTuplizer.buildProxyFactory(PojoEntityTuplizer.java:162)
	at org.hibernate.tuple.entity.AbstractEntityTuplizer.<init>(AbstractEntityTuplizer.java:163)
	at org.hibernate.tuple.entity.PojoEntityTuplizer.<init>(PojoEntityTuplizer.java:58)
	... 131 more
Caused by: java.lang.ClassNotFoundException: java.lang.StackWalker$Option
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.jboss.modules.JDKSpecific.getSystemClass(JDKSpecific.java:183)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:397)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	... 143 more
```

P.S Would it be possible to get a build of the 'master' somewhere? So i don't have to build it? Unfortunately the latest GA release causes a NPE (which has been fixed in a later commit)

--- a/src/main/javassist/util/proxy/DefineClassHelper.java
+++ b/src/main/javassist/util/proxy/DefineClassHelper.java
@@ -51,8 +51,12 @@ public class DefineClassHelper
                 Class<?> defineClass(String name, byte[] b, int off, int len,
                         ClassLoader loader, ProtectionDomain protectionDomain)
                                 throws ClassFormatError {
-                    if (stack.getCallerClass() != SecuredPrivileged.JAVA_9.getClass())
-                        throw new IllegalAccessError("Access denied for caller.");
+                    try {
+                        if (getCallerClass.invoke(stack) != SecuredPrivileged.JAVA_9.getClass())
+                            throw new IllegalAccessError("Access denied for caller.");
+                    } catch (Exception e) {
+                        throw new RuntimeException("cannot initialize", e);
+                    }
                     try {
                         return (Class<?>) defineClass.invokeWithArguments(
                                 sunMiscUnsafeTheUnsafe.theUnsafe,
@@ -64,14 +68,40 @@ public class DefineClassHelper
                     }
                 }
             }
-            private final StackWalker stack = StackWalker
-                    .getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
+            private final Object stack;
+            private final Method getCallerClass;
+            {
+                Class<?> stackWalkerClass = null;
+                try {
+                    stackWalkerClass = Class.forName("java.lang.StackWalker");
+                } catch (ClassNotFoundException e) {
+                    // Skip initialization when the class doesn't exist i.e. we are on JDK < 9
+                }
+                if (stackWalkerClass != null) {
+                    try {
+                        Class<?> optionClass = Class.forName("java.lang.StackWalker$Option");
+                        stack = stackWalkerClass.getMethod("getInstance", optionClass)
+                                // The first one is RETAIN_CLASS_REFERENCE
+                                .invoke(null, optionClass.getEnumConstants()[0]);
+                        getCallerClass = stackWalkerClass.getMethod("getCallerClass");
+                    } catch (Throwable e) {
+                        throw new RuntimeException("cannot initialize", e);
+                    }
+                } else {
+                    stack = null;
+                    getCallerClass = null;
+                }
+            }
             private final ReferencedUnsafe sunMiscUnsafe = getReferencedUnsafe();
             private final ReferencedUnsafe getReferencedUnsafe()
             {
-                if (null != SecuredPrivileged.JAVA_9
-                        && stack.getCallerClass() != this.getClass())
-                    throw new IllegalAccessError("Access denied for caller.");
+                try {
+                    if (null != SecuredPrivileged.JAVA_9
+                            && getCallerClass.invoke(stack) != this.getClass())
+                        throw new IllegalAccessError("Access denied for caller.");
+                } catch (Exception e) {
+                    throw new RuntimeException("cannot initialize", e);
+                }
                 try {
                     SecurityActions.TheUnsafe usf = SecurityActions.getSunMiscUnsafeAnonymously();
                     List<Method> defineClassMethod = usf.methods.get("defineClass");
@@ -91,8 +121,12 @@ public class DefineClassHelper
                     ClassLoader loader, ProtectionDomain protectionDomain)
                             throws ClassFormatError
             {
-                if (stack.getCallerClass() != DefineClassHelper.class)
-                    throw new IllegalAccessError("Access denied for caller.");
+                try {
+                    if (getCallerClass.invoke(stack) != DefineClassHelper.class)
+                        throw new IllegalAccessError("Access denied for caller.");
+                } catch (Exception e) {
+                    throw new RuntimeException("cannot initialize", e);
+                }
                 return sunMiscUnsafe.defineClass(name, b, off, len, loader,
                         protectionDomain);
             }

